pootle/apps/pootle_statistics/models.py

```diff
@@ -207,6 +207,58 @@ class Meta(object):
     # similarity ratio to the result of machine translation
     mt_similarity = models.FloatField(blank=True, null=True)
 
+    @classmethod
+    def record_submissions(cls, unit, old_target, old_state, current_time,
+                           user, store_id, translation_project_id,
+                           submission_type=None):
+        """Records all applicable submissions for `unit`.
+
+        EXTREME HAZARD: this relies on implicit `._<field>_updated` members
+        being available in `unit`. Let's look into replacing such members with
+        something saner (#3895).
+        """
+        create_subs = {}
+
+        # FIXME: extreme implicit hazard
+        if unit._target_updated:
+            create_subs[SubmissionFields.TARGET] = [
+                old_target,
+                unit.target_f,
+            ]
+
+        # FIXME: extreme implicit hazard
+        if unit._state_updated:
+            create_subs[SubmissionFields.STATE] = [
+                old_state,
+                unit.state,
+            ]
+
+        # FIXME: extreme implicit hazard
+        if unit._comment_updated:
+            create_subs[SubmissionFields.COMMENT] = [
+                '',
+                unit.translator_comment or '',
+            ]
+
+        if submission_type is None:
+            submission_type = SubmissionTypes.SYSTEM
```

Comment:
Just checked where this is called and perhaps it is better to keep this, so skip my previous comment.